linter/lints/cabf_br/lint_crl_validity_period.go

```diff
@@ -42,17 +47,73 @@ func (l *crlValidityPeriod) CheckApplies(c *x509.RevocationList) bool {
 }
 
 func (l *crlValidityPeriod) Execute(c *x509.RevocationList) *lint.LintResult {
-	validity := c.NextUpdate.Sub(c.ThisUpdate)
-	if validity <= 0 {
+	/*
+		Let's Encrypt issues CRLs for two distinct purposes:
+		   1) CRLs containing subscriber certificates created by the
+		      crl-updater. These CRLs must have only the distributionPoint and
+		      onlyContainsUserCerts fields set.
+		   2) CRLs containing subordinate CA certificates created by the
+		      ceremony tool. These CRLs must only have the onlyContainsCACerts
+		      field set.
+
+		RFC 5280 Section 5.2.5
+
+		IssuingDistributionPoint ::= SEQUENCE {
+			distributionPoint          [0] DistributionPointName OPTIONAL,
+			onlyContainsUserCerts      [1] BOOLEAN DEFAULT FALSE,
+			onlyContainsCACerts        [2] BOOLEAN DEFAULT FALSE,
+			...
+		}
+	*/
+
+	// The only way to determine which type of CRL we're dealing with, the
+	// issuingDistributionPoint must be parsed and the internal fields
+	// inspected.
+	idpOID := asn1.ObjectIdentifier{2, 5, 29, 28} // id-ce-issuingDistributionPoint
+	idpe := lints.GetExtWithOID(c.Extensions, idpOID)
+	if idpe == nil {
+		return &lint.LintResult{
+			Status:  lint.Warn,
+			Details: "CRL missing IDP",
+		}
+	}
+
+	// Step inside the outer issuingDistributionPoint sequence to get access to
+	// its constituent fields.
+	idpv := cryptobyte.String(idpe.Value)
+	_ = idpv.ReadASN1(&idpv, cryptobyte_asn1.SEQUENCE)
+
+	// We read this boolean as a byte and ensure its value is 0xFF because
+	// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for the
+	// [1] and [2] tagged fields referenced above. For the purposes of this
+	// lint, we just need to know if the field exists.
+	onlyContainsCACerts := make([]byte, 0)
+
+	// Default to subscriber cert CRL.
+	var BRValidity = 10 * 24 * time.Hour
+	var validityString = "10 days"
+
```

Comment:
super-duper-hyper feel free to ignore me nit: I like to declare an object that will be filled by a function call _immediately_ before that call, whenever possible. This has the added benefit of moving your great comment about how we read the boolean to be immediately before the code which actually does that work, rather than separate from it by two blank lines and a wholly unrelated comment.
```suggestion
	// Default to subscriber cert CRL.
	var BRValidity = 10 * 24 * time.Hour
	var validityString = "10 days"
	
	// We read this boolean as a byte and ensure its value is 0xFF because
	// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for the
	// [1] and [2] tagged fields referenced above. For the purposes of this
	// lint, we just need to know if the field exists.
	onlyContainsCACerts := make([]byte, 0)
```